sdk/nestjs/src/DevCycleModule/RequestInterceptor.ts

```diff
@@ -0,0 +1,23 @@
+import {
+    CallHandler,
+    ExecutionContext,
+    Inject,
+    Injectable,
+    NestInterceptor,
+} from '@nestjs/common'
+import { DevCycleClient } from '@devcycle/nodejs-server-sdk'
+
+@Injectable()
+export class RequestInterceptor implements NestInterceptor {
+    constructor(@Inject(DevCycleClient) private client: DevCycleClient) {}
+
+    intercept(
+        context: ExecutionContext,
+        next: CallHandler,
+    ): CallHandler['handle'] {
+        const req = context.switchToHttp().getRequest()
+        req.dvc_client = this.client
```

Comment:
I don't know if it's a problem or not to be adding our own stuff to the request object, but maybe another option is to use Continuous local storage to access the client anywhere in the request context?
https://papooch.github.io/nestjs-cls/introduction/quick-start